### PR TITLE
Add version sanitization regex

### PR DESCRIPTION
**Pull request by schmidt-jake**
Took a stab at addressing #304. Adds a simple regex substitution to sanitize leading zeros out of dependency versions.

Todo:
- [x] add/update unit test(s)

### CODE REVIEW

**Comment by peterdemin** [comment]
A bit of experiments:
```
$ pip install pytz==2022.01
Requirement already satisfied: pytz==2022.01 in .../lib/python3.6/site-packages (2022.1)
$ pip install pytz==2022.00001
Requirement already satisfied: pytz==2022.00001 in .../lib/python3.6/site-packages (2022.1)
```

File: pipcompilemulti/dependency.py [hunk 1]
```diff
@@ -102,7 +102,8 @@ def __init__(self, line):
         regular = self.RE_DEPENDENCY.match(line)
         if regular:
             self.package = regular.group('package')
-            self.version = regular.group('version').strip()
+            _version = regular.group('version').strip()
+            self.version = re.sub(r"(?<=\.)0+(?=\d+)", "", _version)  # strip leading zeros
```
Comment (peterdemin) [inline pipcompilemulti/dependency.py:106]: Looks good. Let's compile the regex once along with other patterns (and lose `+` after `\d`):
```
RE_IGNORED_ZEROS = re.compile(r"(?<=\.)0+(?=\d)")
self.version = self.RE_IGNORED_ZEROS.sub("", regular.group('version').strip())
```

**Review by peterdemin — APPROVED** [review]

File: pipcompilemulti/dependency.py (hunk 1 above)
Comment (schmidt-jake) [reply pipcompilemulti/dependency.py]: Done!

**Comment by peterdemin** [comment]
Thanks!

**Pull request merged**